File: TP_01Afonso_09JoaoGoncalves_15Silvio/DatabaseAccess.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


public class DataBaseAccess
{
    //################################################################################
    //METHOD THAT RETURNS A STRING WITH THE DATABASE CONNECTION
    //note: there may be several SC; choose the one you want to use
    public string ConnectionString()
    {
        string SCLagostimExterna = "Data Source=62.28.39.135,62444;Initial Catalog=ASJregistoTempos;User ID=EFAGONCALVES;Password=123.Abc;";

        //return the one you want to use:
        return SCLagostimExterna;
    }
    //################################################################################
    //METHOD THAT RETURNS A DATATABLE FROM THE PARAMETERS
    // SC (string connection) and ssql (query to the DB).
    public DataTable BuscarDados(string SC, string ssql)
    {
        SqlConnection Con = null;
        try
        {
            Con = new SqlConnection(SC);
            Con.Open();

            SqlCommand cmd = null;
            try
            {
                cmd = new SqlCommand(ssql, Con);
                SqlDataAdapter adapter = null;
                try
                {
                    adapter = new SqlDataAdapter(cmd);
                    DataTable dt = new DataTable();
                    adapter.Fill(dt);
                    return dt;
                }
                finally
                {
                    adapter?.Dispose();
                }
            }
            finally
            {
                cmd?.Dispose();
            }
        }
        finally
        {
            Con?.Dispose();
        }
    }
}

File: P2AfonsoFelippe/DataBaseAccess.vb
Imports System.Data.SqlClient

Public Class DataBaseAccess

    Dim strConn As String = "Data Source=62.28.39.135,62444;Initial Catalog=bdFelippeAfonso_Produtos;User ID=EFAFELIPPE;Password=123.Abc##;"

    Public Function BuscarDados(ssql As String) As DataTable

        Dim Con As SqlConnection = New SqlConnection(strConn)
        Con.Open()

        Dim cmd As SqlCommand = New SqlCommand(ssql, Con)

        Dim adapter As SqlDataAdapter = New SqlDataAdapter(cmd)

        Dim dt As New DataTable()
        adapter.Fill(dt)

        adapter.Dispose()
        Con.Close()

        Return dt

    End Function

End Class
